plugins/woocommerce-blocks/assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -264,7 +269,7 @@ const productGallery = {
 			);
 
 			const selectFirstImage = () => {
-				context.selectedImage = context.firstMainImageId;
+				selectImage( context, 'first' );
```

Comment:
`first` needs updating here I think

Reply:
Added a `first` as accepted option, forgot it's still in use in here. Also, as we discussed elsewhere I'll double check why changing images with variation change doesn't work.

https://github.com/woocommerce/woocommerce/pull/54570/commits/41f56ee7ca9e57c5f11fd04aea5b4f477f6f7e8c

Reply:
Ok, I verified and variable product images change as on `trunk` which means:
- change the image when it's in available images (it won't be changed if it's only available on dialog but we'll fix it)
- change the image to first upon clearing.

So I guess we're good to go with this PR.

https://github.com/user-attachments/assets/56e5033b-5644-4354-9f6a-2a6b7884f229